test/cri-containerd/logging_binary_test.go

```diff
@@ -0,0 +1,178 @@
+				pullRequiredLcowImages(t, requiredImages)
+			} else {
+				pullRequiredImages(t, requiredImages)
+			}
+
+			podReq := getRunPodSandboxRequest(t, test.runtimeHandler)
+			podID := runPodSandbox(t, client, ctx, podReq)
+			defer removePodSandbox(t, client, ctx, podID)
+
+			logFileName := fmt.Sprintf("%s/stdout-%s.txt", logBinaryRoot, test.name)
+			conReq := getCreateContainerRequest(podID, test.containerName, test.containerImage, test.cmd, podReq.Config)
+			conReq.Config.LogPath = logPath + fmt.Sprintf("?%s", logFileName)
+
+			createAndRunContainer(t, client, ctx, conReq)
+
+			if _, err := os.Stat(logFileName); os.IsNotExist(err) {
+				t.Fatalf("log file was not created: %s", logFileName)
+			}
+			defer os.Remove(logFileName)
+
+			ok, err := assertFileContent(logFileName, test.expectedContent)
+			if err != nil {
+				t.Fatalf("failed to read log file: %s", err)
+			}
+
+			if !ok {
+				t.Fatalf("file content validation failed: %s", test.expectedContent)
+			}
+		})
+	}
+
+	// Negative tests
+	for _, test := range tests {
+		t.Run(test.name+"_Negative", func(t *testing.T) {
+			requireFeatures(t, test.requiredFeatures...)
+
+			requiredImages := []string{test.sandboxImage, test.containerImage}
+			if test.runtimeHandler == lcowRuntimeHandler {
+				pullRequiredLcowImages(t, requiredImages)
+			} else {
+				pullRequiredImages(t, requiredImages)
+			}
+
+			podReq := getRunPodSandboxRequest(t, test.runtimeHandler)
+			podID := runPodSandbox(t, client, ctx, podReq)
+			defer removePodSandbox(t, client, ctx, podID)
+
+			nonExistentPath := "/does/not/exist/log.txt"
+			conReq := getCreateContainerRequest(podID, test.containerName, test.containerImage, test.cmd, podReq.Config)
+			conReq.Config.LogPath = logPath + fmt.Sprintf("?%s", nonExistentPath)
+
+			containerID := createContainer(t, client, ctx, conReq)
+			defer removeContainer(t, client, ctx, containerID)
+
+			// This should fail, since the filepath doesn't exist
+			_, err := client.StartContainer(ctx, &runtime.StartContainerRequest{
+				ContainerId: containerID,
+			})
+			if err == nil {
+				t.Fatal("container start should fail")
+			}
+
+			if !strings.Contains(err.Error(), "failed to start binary logger") {
+				t.Fatalf("expected 'failed to start binary logger' error, got: %s", err)
+			}
+		})
+	}
+}
+
+func createAndRunContainer(t *testing.T, client runtime.RuntimeServiceClient, ctx context.Context, conReq *runtime.CreateContainerRequest) {
+	containerID := createContainer(t, client, ctx, conReq)
+	defer removeContainer(t, client, ctx, containerID)
+
+	startContainer(t, client, ctx, containerID)
+	defer stopContainer(t, client, ctx, containerID)
+
+	// Let stdio kick in
+	time.Sleep(time.Second * 1)
+}
+
+func assertFileContent(path string, content string) (bool, error) {
+	fileContent, err := ioutil.ReadFile(path)
+	if err != nil {
+		return false, err
+	}
+
+	return strings.Contains(string(fileContent), content), nil
```

Comment:
Do we want *just* contains here?

Reply:
yeah, I think it's enough to validate that things work. I didn't want to add too many hardcoded strings and do exact matches and all that